tests/e2e/utils/service_utils.go

```diff
@@ -0,0 +1,82 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+const (
+	ServiceAnnotationDNSLabelName = "service.beta.kubernetes.io/azure-dns-label-name"
+)
+
+// DefaultService returns a default service representation
+func DefaultService(name string) (result *v1.Service, labels map[string]string) {
+	result = &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: labels,
+		},
+	}
+	return
+}
+
+// CreateLoadBalancerService creates a new service with service port
+func CreateLoadBalancerService(c clientset.Interface, name string, labels map[string]string, namespace string, ports []v1.ServicePort) (*v1.Service, error) {
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Annotations: map[string]string{
+				ServiceAnnotationDNSLabelName: name + string(uuid.NewUUID()),
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Selector: labels,
+			Ports:    ports,
+			Type:     "LoadBalancer",
+		},
+	}
+	return c.CoreV1().Services(namespace).Create(service)
+}
+
+func getServiceDNS(service *v1.Service) string {
+	region := ExtractRegion()
+	return service.Annotations[ServiceAnnotationDNSLabelName] + fmt.Sprintf(".%s.cloudapp.azure.com", region)
+}
+
+// WaitExternalDNS returns ip of ingress
+func WaitExternalDNS(c clientset.Interface, namespace string, name string) (string, error) {
```

Comment:
This function could only wait till service is ready, the dns label is specified by user thus should be known by user